bin/rad-patch

```diff
@@ -189,162 +189,31 @@
-        "--------\n\n"
-        (string/unlines (map pretty-comment cs))))))
-
-(def comment-divider "----------------------------------------")
-
-(def add-comment-md
-  (fn [x]
-    (string-append
-     "\n" comment-divider "\n"
-     "---Add comment above the line.\n\n"
-     x)))
-
 (def add-read-status!
   (fn [machine i]
     (insert :read? (newness/read-since-modified!? machine i) i)))
 
-(def list-items
-  "Lists all patches and filters them by `options`. If `--new` is passed as an
-  option only recent items are shown. If `--fancy is passed, the list is opened
-  via fzf."
-  (fn [machine options]
-
-    (def mk-key
-      (fn [patch]
-        (if (lookup :fancy options)
-          [(pretty-headline patch) patch]
-          [(pretty-row patch) patch])))
+(def add-username
+  (fn [d]
+    (def n (lookup-default :author-name "???" d))
+    (insert :username n d)))
 
-    (def state-filt
-      (match (map (fn [opt] (read (string-append ":" opt))) (lookup :state options))
-        []      (fn [x] #t)
-        'args   (fn [x] (elem? (lookup :state x) args))))
-
-    (def patches (values (list-patches machine)))
+(def add-username-patch
+  (fn [i]
+    (add-username
+     (insert :comments (map add-username (lookup :comments i)) i))))
 
-    (def newness-filt
-      (if (lookup :unread options)
-        (fn [x] (not (lookup :read? x)))
-        (fn [x] #t)))
-
-  (def items
-    (map mk-key
-         (filter (and-predicate state-filt newness-filt)
-                 (map (fn [i] (add-read-status! machine i))
-                      (reverse patches)))))
-
-    (def items-dict (dict-from-seq items))
-    (if (empty-seq? items)
-      (put-str! "No patches!")
-      (if (lookup :fancy options)
-        (match (fzf-select-with-preview!
-                 (map first items)
-                 (fn [s] (pretty-patch (lookup s items-dict))))
-               (/just 's)
-               (do (def selected (lookup s items-dict))
-                   (def patch-number (match selected {:number 'n} n))
-                   (def pretty-item
-                     (add-comment-md
-                       (pretty-patch selected)))
-                   (def edited-item (edit-in-editor! pretty-item))
-                   (def added-comment
-                     (string/unlines
-                      (take-while (fn [x] (not (eq? x comment-divider)))
-                                  (string/lines edited-item))))
-                   (put-str!
-                    (if (eq? added-comment "")
-                      "Item was not updated."
-                      (do (simple-add-comment! machine patch-number added-comment (get-git-username!))
-                          (string-append "Sent comment: " added-comment)))))
-               _ (put-str! "No selection made."))
-         (do
-           (put-str! (pretty-table-header))
-           (map (fn [d] (put-str! (first d))) items)
-           (put-str! (items/summary "patches" patches)))))))
+(def enrich-patch
```

Comment:
Similarly I think `enrich-patch` and `enrich-issue` can just be moved to `enrich-item` in `items.rad`.